app/models/claim/cleaner.rb

```diff
@@ -0,0 +1,46 @@
+class Claim
+  class Cleaner
+    attr_reader :claims
+
+    PII_ATTRIBUTES_TO_DELETE = [
+      :first_name,
+      :middle_name,
+      :surname,
+      :date_of_birth,
+      :address_line_1,
+      :address_line_2,
+      :address_line_3,
+      :address_line_4,
+      :postcode,
+      :payroll_gender,
+      :national_insurance_number,
+      :bank_sort_code,
+      :bank_account_number,
+      :building_society_roll_number,
+    ]
+
+    MINIMUM_DATE_OF_LAST_ACTION = 2.months.ago
+
+    def delete_pii
+      old_claims_rejected_or_paid.update_all(attribute_values_to_set)
+    end
+
+    private
+
+    def attribute_values_to_set
+      PII_ATTRIBUTES_TO_DELETE.map { |attr| [attr, nil] }.to_h.merge(
+        pii_removed_at: Time.now
+      )
+    end
+
+    def old_claims_rejected_or_paid
```

Comment:
I'm wondering if, with this, we could be more specific about how we fetch the results, to guard against any changes being made to the `Decision`s enum. Something like:

```ruby
Claim.left_outer_joins(payment: [:payroll_run])
  .joins(:decision)
  .where(pii_removed_at: nil)
  .where(
    "(decisions.result = :rejected AND claims.updated_at < :minimum_time) OR scheduled_payment_date < :minimum_time",
    minimum_time: MINIMUM_DATE_OF_LAST_ACTION,
    rejected: Decision.results.fetch(:rejected)
  )
```

(using `fetch` here would raise an error if we change the name of the `rejected` enum, and alert us that there was something wrong)